### ISSUE TITLE
Arabic in Manim

### ISSUE THREAD

**Issue by AyoubRahbawi** [3b1b/manim#1173]
hello guys i did try this solution here https://github.com/3b1b/manim/issues/342  i added the usepackages of arabic as it said there .  but i got random arabic letter instead of english and and also the arabic text i wanted was not written correctly please i need some help here .

**Comment by Tony031218** [comment]
Try to use `Text` instead of `Tex(t)Mobject` ?

**Comment by yousif-hag-ahmed** [comment]
Try this
https://github.com/3b1b/manim/issues/1241#issuecomment-753416291

**Comment by mohamed99akram** [comment]
You can write your text "مرحبا" [here](http://www.arabic-keyboard.org/photoshop-arabic/) and copy the output. then use Text("ﺎﺒﺣﺮﻣ") in your code. 
it works for me, the text is written correctly.
There are 2 things I still need to know though:
 1- How to change the font
 2- How to write(text) from right to left instead of left to write

**Comment by yousif-hag-ahmed** [comment]
I got it working
please check this ;

![image](https://user-images.githubusercontent.com/69925471/110695612-1ae34680-81b8-11eb-958f-ef35460b89c7.png)

use this tex file changes 

\documentclass[preview]{standalone}



\usepackage[utf8]{inputenc}
\usepackage[arabic,english]{babel}
\usepackage{arabtex}
\usepackage[LAE,LFE]{fontenc}


\usepackage[T1]{fontenc}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{dsfont}
\usepackage{setspace}
\usepackage{tipa}
\usepackage{relsize}
\usepackage{textcomp}
\usepackage{mathrsfs}
\usepackage{calligra}
\usepackage{wasysym}
\usepackage{ragged2e}
\usepackage{physics}
\usepackage{xcolor}
\usepackage{microtype}
%\DisableLigatures{encoding = *, family = * }
\usepackage[UTF8]{ctex}
\linespread{1}

\begin{document}

\begin{align*}
\quad
\end{align*}

\end{document}





and call using 
TextMobject
like this 
example_text = TextMobject(
            "هذه بعض المعادلات ",
            tex_to_color_map={"text": YELLOW}
        )


hope this helps 👍

**Comment by Sameed123** [comment]
> I got it working
> please check this ;
> 
> ![image](https://user-images.githubusercontent.com/69925471/110695612-1ae34680-81b8-11eb-958f-ef35460b89c7.png)
> 
> use this tex file changes
> 
> \documentclass[preview]{standalone}
> 
> \usepackage[utf8]{inputenc}
> \usepackage[arabic,english]{babel}
> \usepackage{arabtex}
> \usepackage[LAE,LFE]{fontenc}
> 
> \usepackage[T1]{fontenc}
> \usepackage{amsmath}
> \usepackage{amssymb}
> \usepackage{dsfont}
> \usepackage{setspace}
> \usepackage{tipa}
> \usepackage{relsize}
> \usepackage{textcomp}
> \usepackage{mathrsfs}
> \usepackage{calligra}
> \usepackage{wasysym}
> \usepackage{ragged2e}
> \usepackage{physics}
> \usepackage{xcolor}
> \usepackage{microtype}
> %\DisableLigatures{encoding = *, family = * }
> \usepackage[UTF8]{ctex}
> \linespread{1}
> 
> \begin{document}
> 
> \begin{align*}
> \quad
> \end{align*}
> 
> \end{document}
> 
> and call using
> TextMobject
> like this
> example_text = TextMobject(
> "هذه بعض المعادلات ",
> tex_to_color_map={"text": YELLOW}
> )
> 
> hope this helps 👍
> 
> please note : after this changes you need to use Text for writing in English

Hi Yousif,
I tried changing my tex_template.tex file according as you have described.

For some reason I am getting an error.
It's saying: "Latex error converting to dvi."

Do you know why this is happening.

Maybe it's some issue with my version of manim.

May I know which version you are using?

Also, I am using a Mac. If you are not using Mac then maybe that's why it is not working for me...
(I am mostly a noob at this sort of stuff so excuse me if my questions are too stupid...)

Also, I am running manim on pycharm. Can this possibly be a reason for this error?


Basically I want to know exactly how you are using manim and which version it is so that I can try to do exactly what you are doing.

Any help here would be great.

Thanks a lot.

**Comment by yousif-hag-ahmed** [comment]
hi 
I am using Windows actually , can you post your py code so that I can take a look at it and maybe try it here and see .
thanks

**Comment by mahdirezaie336** [comment]
> I got it working
> please check this ;
> 
> ![image](https://user-images.githubusercontent.com/69925471/110695612-1ae34680-81b8-11eb-958f-ef35460b89c7.png)
> 
> use this tex file changes
> 
> \documentclass[preview]{standalone}
> 
> \usepackage[utf8]{inputenc}
> \usepackage[arabic,english]{babel}
> \usepackage{arabtex}
> \usepackage[LAE,LFE]{fontenc}
> 
> \usepackage[T1]{fontenc}
> \usepackage{amsmath}
> \usepackage{amssymb}
> \usepackage{dsfont}
> \usepackage{setspace}
> \usepackage{tipa}
> \usepackage{relsize}
> \usepackage{textcomp}
> \usepackage{mathrsfs}
> \usepackage{calligra}
> \usepackage{wasysym}
> \usepackage{ragged2e}
> \usepackage{physics}
> \usepackage{xcolor}
> \usepackage{microtype}
> %\DisableLigatures{encoding = *, family = * }
> \usepackage[UTF8]{ctex}
> \linespread{1}
> 
> \begin{document}
> 
> \begin{align*}
> \quad
> \end{align*}
> 
> \end{document}
> 
> and call using
> TextMobject
> like this
> example_text = TextMobject(
> "هذه بعض المعادلات ",
> tex_to_color_map={"text": YELLOW}
> )
> 
> hope this helps 
> 
> please note : after this changes you need to use Text for writing in English

Hi yousif
I did exactly what you said but for me it gives this exception:

Traceback (most recent call last):
  File "/home/mahdi/.local/lib/python3.9/site-packages/manimlib/extract_scene.py", line 155, in main
    scene = SceneClass(**scene_kwargs)
  File "/home/mahdi/.local/lib/python3.9/site-packages/manimlib/scene/scene.py", line 75, in __init__
    self.construct()
  File "main.py", line 7, in construct
    my_text = TextMobject("هذه بعض المعادلات ", tex_to_color_map={"text": YELLOW})
  File "/home/mahdi/.local/lib/python3.9/site-packages/manimlib/mobject/svg/tex_mobject.py", line 150, in __init__
    self.break_up_by_substrings()
  File "/home/mahdi/.local/lib/python3.9/site-packages/manimlib/mobject/svg/tex_mobject.py", line 190, in break_up_by_substrings
    sub_tex_mob.move_to(self.submobjects[last_submob_index], RIGHT)
IndexError: list index out of range


Do you have any solutions?

**Comment by seloumi** [comment]
Here how can you write arabic with mixed math, you need to use lualatex with a recent tex distribution

```python
from manim import *

my_template = TexTemplate(
tex_compiler="lualatex",
preamble=r"""
\usepackage[bidi = basic]{babel}
\babelprovide[main, import]{arabic}
\babelfont{rm}{Amiri}""")

class ArabTex(Tex):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, tex_template=my_template, **kwargs)

class Test(Scene):
    def construct(self):
        arab_text= ArabTex(r'نص عربي طويل مع عبارات رياضية $ax+b$',font_size=50).to_corner(UR)
        self.play(Write(arab_text))
        self.wait()
        tex = MathTex(r'f(x) &= 3 + 2 + 1\\ &= 5 + 1 \\ &= 6', font_size=50)
        tex.next_to(arab_text, DOWN,buff=0.5)
        self.add(tex)
        self.wait()
        circle= Circle()
        circle.next_to(tex, DOWN, buff=0.5) 
        self.play(Create(circle))
```

https://user-images.githubusercontent.com/8249073/236542312-466a1bcf-5f16-447a-86e3-f7143db234e1.mp4

**Comment by Xenomorphs** [comment]
> Here how can you write arabic with mixed math, you need to use lualatex with a recent tex distribution
> 
> ```python
> from manim import *
> 
> my_template = TexTemplate(
> tex_compiler="lualatex",
> preamble=r"""
> \usepackage[bidi = basic]{babel}
> \babelprovide[main, import]{arabic}
> \babelfont{rm}{Amiri}""")
> 
> class ArabTex(Tex):
>     def __init__(self, *args, **kwargs):
>         super().__init__(*args, tex_template=my_template, **kwargs)
> 
> class Test(Scene):
>     def construct(self):
>         arab_text= ArabTex(r'نص عربي طويل مع عبارات رياضية $ax+b$',font_size=50).to_corner(UR)
>         self.play(Write(arab_text))
>         self.wait()
>         tex = MathTex(r'f(x) &= 3 + 2 + 1\\ &= 5 + 1 \\ &= 6', font_size=50)
>         tex.next_to(arab_text, DOWN,buff=0.5)
>         self.add(tex)
>         self.wait()
>         circle= Circle()
>         circle.next_to(tex, DOWN, buff=0.5) 
>         self.play(Create(circle))
> ```
> 
>  Test.mp4

I'm getting this error, any way to fix it?:

[06/16/23 22:32:30] INFO     Writing نص عربي طويل مع عبارات رياضية    tex_file_writing.py:97                             $ax+b$ to media\Tex\96664e1c55cbdbda.tex
[06/16/23 22:32:37] ERROR    LaTeX compilation error: LaTeX Error:   tex_file_writing.py:285                             File `luatex85.sty' not found.

                    ERROR    LaTeX compilation error: Emergency      tex_file_writing.py:285                             stop.

                    ERROR    LaTeX compilation error:  ==> Fatal     tex_file_writing.py:285                             error occurred, bad output DVI file
                             produced!

╭─────────────────────────── Traceback (most recent call last) ────────────────────────────╮│ C:\tools\Manim\Lib\site-packages\manim\cli\render\commands.py:115 in render              ││                                                                                          ││   112 │   │   │   try:                                                                   ││   113 │   │   │   │   with tempconfig({}):                                               ││   114 │   │   │   │   │   scene = SceneClass()                                           ││ ❱ 115 │   │   │   │   │   scene.render()                                                 ││   116 │   │   │   except Exception:                                                      ││   117 │   │   │   │   error_console.print_exception()                                    ││   118 │   │   │   │   sys.exit(1)                                                        ││                                                                                          ││ C:\tools\Manim\Lib\site-packages\manim\scene\scene.py:223 in render                      ││                                                                                          ││    220 │   │   """                                                                       ││    221 │   │   self.setup()                                                              ││    222 │   │   try:                                                                      ││ ❱  223 │   │   │   self.construct()                                                      ││    224 │   │   except EndSceneEarlyException:                                            ││    225 │   │   │   pass                                                                  ││    226 │   │   except RerunSceneException as e:                                          ││                                                                                          ││ D:\For Manim and Node\project\scene.py:16 in construct                                   ││                                                                                          ││   13                                                                                     ││   14 class Test(Scene):                                                                  ││   15 │   def construct(self):                                                            ││ ❱ 16 │   │   arab_text= ArabTex(r'نص عربي طويل مع عبارات رياضية $ax+b$',font_size=50).to ││   17 │   │   self.play(Write(arab_text))                                                 ││   18 │   │   self.wait()                                                                 ││   19 │   │   tex = MathTex(r'f(x) &= 3 + 2 + 1\\ &= 5 + 1 \\ &= 6', font_size=50)        ││                                                                                          ││ D:\For Manim and Node\project\scene.py:12 in __init__                                    ││                                                                                          ││    9                                                                                     ││   10 class ArabTex(Tex):                                                                 ││   11 │   def __init__(self, *args, **kwargs):                                            ││ ❱ 12 │   │   super().__init__(*args, tex_template=my_template, **kwargs)                 ││   13                                                                                     ││   14 class Test(Scene):                                                                  ││   15 │   def construct(self):                                                            ││                                                                                          ││ C:\tools\Manim\Lib\site-packages\manim\mobject\text\tex_mobject.py:449 in __init__       ││                                                                                          ││   446 │   def __init__(                                                                  ││   447 │   │   self, *tex_strings, arg_separator="", tex_environment="center", **kwargs   ││   448 │   ):                                                                             ││ ❱ 449 │   │   super().__init__(                                                          ││   450 │   │   │   *tex_strings,                                                          ││   451 │   │   │   arg_separator=arg_separator,                                           ││   452 │   │   │   tex_environment=tex_environment,                                       ││                                                                                          ││ C:\tools\Manim\Lib\site-packages\manim\mobject\text\tex_mobject.py:274 in __init__       ││                                                                                          ││   271 │   │   self.brace_notation_split_occurred = False                                 ││   272 │   │   self.tex_strings = self._break_up_tex_strings(tex_strings)                 ││   273 │   │   try:                                                                       ││ ❱ 274 │   │   │   super().__init__(                                                      ││   275 │   │   │   │   self.arg_separator.join(self.tex_strings),                         ││   276 │   │   │   │   tex_environment=self.tex_environment,                              ││   277 │   │   │   │   tex_template=self.tex_template,                                    ││                                                                                          ││ C:\tools\Manim\Lib\site-packages\manim\mobject\text\tex_mobject.py:83 in __init__        ││                                                                                          ││    80 │   │                                                                              ││    81 │   │   assert isinstance(tex_string, str)                                         ││    82 │   │   self.tex_string = tex_string                                               ││ ❱  83 │   │   file_name = tex_to_svg_file(                                               ││    84 │   │   │   self._get_modified_expression(tex_string),                             ││    85 │   │   │   environment=self.tex_environment,                                      ││    86 │   │   │   tex_template=self.tex_template,                                        ││                                                                                          ││ C:\tools\Manim\Lib\site-packages\manim\utils\tex_file_writing.py:54 in tex_to_svg_file   ││                                                                                          ││    51 │   if tex_template is None:                                                       ││    52 │   │   tex_template = config["tex_template"]                                      ││    53 │   tex_file = generate_tex_file(expression, environment, tex_template)            ││ ❱  54 │   dvi_file = compile_tex(                                                        ││    55 │   │   tex_file,                                                                  ││    56 │   │   tex_template.tex_compiler,                                                 ││    57 │   │   tex_template.output_format,                                                ││                                                                                          ││ C:\tools\Manim\Lib\site-packages\manim\utils\tex_file_writing.py:200 in compile_tex      ││                                                                                          ││   197 │   │   exit_code = os.system(command)                                             ││   198 │   │   if exit_code != 0:                                                         ││   199 │   │   │   log_file = tex_file.with_suffix(".log")                                ││ ❱ 200 │   │   │   print_all_tex_errors(log_file, tex_compiler, tex_file)                 ││   201 │   │   │   raise ValueError(                                                      ││   202 │   │   │   │   f"{tex_compiler} error converting to"                              ││   203 │   │   │   │   f" {output_format[1:]}. See log output above or"                   ││                                                                                          ││ C:\tools\Manim\Lib\site-packages\manim\utils\tex_file_writing.py:269 in                  ││ print_all_tex_errors                                                                     ││                                                                                          ││   266 │   │   with tex_file.open() as f:                                                 ││   267 │   │   │   tex = f.readlines()                                                    ││   268 │   │   for error_index in error_indices:                                          ││ ❱ 269 │   │   │   print_tex_error(tex_compilation_log, error_index, tex)                 ││   270                                                                                    ││   271                                                                                    ││   272 LATEX_ERROR_INSIGHTS = [                                                           ││                                                                                          ││ C:\tools\Manim\Lib\site-packages\manim\utils\tex_file_writing.py:293 in print_tex_error  ││                                                                                          ││   290 │   # failure. This code finds the next such line after the error current error me ││   291 │   line_of_tex_error = (                                                          ││   292 │   │   int(                                                                       ││ ❱ 293 │   │   │   [                                                                      ││   294 │   │   │   │   log_line                                                           ││   295 │   │   │   │   for log_line in tex_compilation_log[error_start_index:]            ││   296 │   │   │   │   if log_line.startswith("l.")                                       │╰──────────────────────────────────────────────────────────────────────────────────────────╯IndexError: list index out of range

**Comment by Xenomorphs** [comment]
> Here how can you write arabic with mixed math, you need to use lualatex with a recent tex distribution
> 
> ```python
> from manim import *
> 
> my_template = TexTemplate(
> tex_compiler="lualatex",
> preamble=r"""
> \usepackage[bidi = basic]{babel}
> \babelprovide[main, import]{arabic}
> \babelfont{rm}{Amiri}""")
> 
> class ArabTex(Tex):
>     def __init__(self, *args, **kwargs):
>         super().__init__(*args, tex_template=my_template, **kwargs)
> 
> class Test(Scene):
>     def construct(self):
>         arab_text= ArabTex(r'نص عربي طويل مع عبارات رياضية $ax+b$',font_size=50).to_corner(UR)
>         self.play(Write(arab_text))
>         self.wait()
>         tex = MathTex(r'f(x) &= 3 + 2 + 1\\ &= 5 + 1 \\ &= 6', font_size=50)
>         tex.next_to(arab_text, DOWN,buff=0.5)
>         self.add(tex)
>         self.wait()
>         circle= Circle()
>         circle.next_to(tex, DOWN, buff=0.5) 
>         self.play(Create(circle))
> ```
> 
>  Test.mp4

I'm getting this error, do you know why?:

```
[06/16/23 22:32:30] INFO     Writing نص عربي طويل مع عبارات رياضية    tex_file_writing.py:97                             $ax+b$ to media\Tex\96664e1c55cbdbda.tex
[06/16/23 22:32:37] ERROR    LaTeX compilation error: LaTeX Error:   tex_file_writing.py:285                             File 'luatex85.sty' not found.

ERROR    LaTeX compilation error: Emergency  stop.    tex_file_writing.py:285                             

ERROR    LaTeX compilation error:  ==> Fatal error occured, bad output DVI file produced!  tex_file_writing.py:285             
```

**Comment by Xenomorphs** [comment]
> > I got it working
> > please check this ;
> > ![image](https://user-images.githubusercontent.com/69925471/110695612-1ae34680-81b8-11eb-958f-ef35460b89c7.png)
> > use this tex file changes
> > \documentclass[preview]{standalone}
> > \usepackage[utf8]{inputenc}
> > \usepackage[arabic,english]{babel}
> > \usepackage{arabtex}
> > \usepackage[LAE,LFE]{fontenc}
> > \usepackage[T1]{fontenc}
> > \usepackage{amsmath}
> > \usepackage{amssymb}
> > \usepackage{dsfont}
> > \usepackage{setspace}
> > \usepackage{tipa}
> > \usepackage{relsize}
> > \usepackage{textcomp}
> > \usepackage{mathrsfs}
> > \usepackage{calligra}
> > \usepackage{wasysym}
> > \usepackage{ragged2e}
> > \usepackage{physics}
> > \usepackage{xcolor}
> > \usepackage{microtype}
> > %\DisableLigatures{encoding = *, family = * }
> > \usepackage[UTF8]{ctex}
> > \linespread{1}
> > \begin{document}
> > \begin{align*}
> > \quad
> > \end{align*}
> > \end{document}
> > and call using
> > TextMobject
> > like this
> > example_text = TextMobject(
> > "هذه بعض المعادلات ",
> > tex_to_color_map={"text": YELLOW}
> > )
> > hope this helps 👍
> > please note : after this changes you need to use Text for writing in English
> 
> Hi Yousif, I tried changing my tex_template.tex file according as you have described.
> 
> For some reason I am getting an error. It's saying: "Latex error converting to dvi."
> 
> Do you know why this is happening.
> 
> Maybe it's some issue with my version of manim.
> 
> May I know which version you are using?
> 
> Also, I am using a Mac. If you are not using Mac then maybe that's why it is not working for me... (I am mostly a noob at this sort of stuff so excuse me if my questions are too stupid...)
> 
> Also, I am running manim on pycharm. Can this possibly be a reason for this error?
> 
> Basically I want to know exactly how you are using manim and which version it is so that I can try to do exactly what you are doing.
> 
> Any help here would be great.
> 
> Thanks a lot.

Getting the same error on Windows. Did you find a way to write Arabic?

**Comment by Xenomorphs** [comment]
> Here how can you write arabic with mixed math, you need to use lualatex with a recent tex distribution
> 
> ```python
> from manim import *
> 
> my_template = TexTemplate(
> tex_compiler="lualatex",
> preamble=r"""
> \usepackage[bidi = basic]{babel}
> \babelprovide[main, import]{arabic}
> \babelfont{rm}{Amiri}""")
> 
> class ArabTex(Tex):
>     def __init__(self, *args, **kwargs):
>         super().__init__(*args, tex_template=my_template, **kwargs)
> 
> class Test(Scene):
>     def construct(self):
>         arab_text= ArabTex(r'نص عربي طويل مع عبارات رياضية $ax+b$',font_size=50).to_corner(UR)
>         self.play(Write(arab_text))
>         self.wait()
>         tex = MathTex(r'f(x) &= 3 + 2 + 1\\ &= 5 + 1 \\ &= 6', font_size=50)
>         tex.next_to(arab_text, DOWN,buff=0.5)
>         self.add(tex)
>         self.wait()
>         circle= Circle()
>         circle.next_to(tex, DOWN, buff=0.5) 
>         self.play(Create(circle))
> ```
> 
>  Test.mp4

Can you also please tell me how you've set everything up for it to work? Please give as much detail as possible, thanks.

**Comment by seloumi** [comment]
@Xenomorphs  what tex distribution you have on your system?

**Comment by halakilm** [comment]
```python
from manim import *
from manim import config

# إعدادات الفيديو لجودة 4K (2160p) للفيديوهات العمودية
config.pixel_height = 3840  # الارتفاع للبكسل لتناسب 4K
config.pixel_width = 2160   # العرض للبكسل لنسبة 9:16
config.frame_rate = 60      # معدل الإطارات

class HelloWorld(Scene):
    def construct(self):
        # انشاء النص مع حجم خط أكبر
        النص_بالعربية = Text(
            " .{وَلَا تَحْسَبَنَّ الَّذِينَ قُتِلُوا فِي سَبِيلِ اللَّهِ أَمْوَاتًا بَلْ أَحْيَاءٌ عِنْدَ رَبِّهِمْ يُرْزَقُونَ}.",
            font="Times New Roman", color=WHITE, font_size=96).scale(1.5)[::-1]
        النص_بالعربية.to_edge(UP)
        # عرض النص
        self.play(Write(النص_بالعربية))
        self.wait(2)

```

**Comment by halakilm** [comment]
```python
from manim import *
from manim import config

# Configurer Manim pour produire des vidéos en qualité 2160p à 60fps
config.pixel_height = 2160
config.pixel_width = 3840
config.frame_rate = 60
class HelloWorld(Scene):
    def construct(self):
        # انشاء النص مع حجم خط أكبر
        النص_بالعربية = Text(
            " .{وَلَا تَحْسَبَنَّ الَّذِينَ قُتِلُوا فِي سَبِيلِ اللَّهِ أَمْوَاتًا بَلْ أَحْيَاءٌ عِنْدَ رَبِّهِمْ يُرْزَقُونَ}.",
            font="Times New Roman", color=WHITE, font_size=25).scale(1.5)[::-1]

        # عرض النص
        self.play(Write(النص_بالعربية))
        self.wait(2)

```

**Comment by halakilm** [comment]
https://github.com/user-attachments/assets/1343d282-0817-4ac8-b0c1-b940f5fe34bb

**Comment by hojjatjafary** [comment]
> ```python
> from manim import *
> from manim import config
> 
> # Configurer Manim pour produire des vidéos en qualité 2160p à 60fps
> config.pixel_height = 2160
> config.pixel_width = 3840
> config.frame_rate = 60
> class HelloWorld(Scene):
>     def construct(self):
>         # انشاء النص مع حجم خط أكبر
>         النص_بالعربية = Text(
>             " .{وَلَا تَحْسَبَنَّ الَّذِينَ قُتِلُوا فِي سَبِيلِ اللَّهِ أَمْوَاتًا بَلْ أَحْيَاءٌ عِنْدَ رَبِّهِمْ يُرْزَقُونَ}.",
>             font="Times New Roman", color=WHITE, font_size=25).scale(1.5)[::-1]
> 
>         # عرض النص
>         self.play(Write(النص_بالعربية))
>         self.wait(2)
> ```

This works but does not have good support when Arabic and English texts are combined.

**Comment by seloumi** [comment]
I don't know what changed from last year, the code was working fine but now the Arabic text is not showing !?

```
from manim import *

my_template = TexTemplate(
tex_compiler="lualatex",
preamble=r"""
\usepackage[bidi = basic]{babel}
\babelprovide[main, import]{arabic}
\babelfont{rm}{Amiri}""")

class ArabTex(Tex):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, tex_template=my_template, **kwargs)

class Test(Scene):
    def construct(self):
        arab_text= ArabTex(r'نص عربي طويل مع عبارات رياضية $ax+b$',font_size=50).to_corner(UR)
        self.add(arab_text)
```
